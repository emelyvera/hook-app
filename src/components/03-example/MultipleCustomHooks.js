import React from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import '../02-useEffect/effects.css'

export const MultipleCustomHooks = () => {

    const { counter, increment } = useCounter(1);

    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);

    /**
     * * cuando data no ha cargado es null y la negacin de null es true y 
     * * !!null es false y data en 0 cumple la condicion 
     * * si existed la data (!!data) entonces (&&) extrae la pocision 0 de la data (data[0])
     */
    const { author, quote } = !!data && data[0];

    return (
        <>
            <h1>BreakingBad Quotes</h1>
            <hr />

            {
                loading
                    ? (
                        <div className='alert alert-info text-center'>
                            Loading...
                        </div>

                    )
                    : (
                        <blockquote className='blockquote text-rigth'>
                            <p className='mb--0'>{quote}</p>
                            <footer className='blockquote-footer'>{author}</footer>
                        </blockquote>

                    )
            }

            <button
                className='btn btn-primary'
                onClick={increment}>
                    Siguiente quote
            </button>

        </>

    )
};
