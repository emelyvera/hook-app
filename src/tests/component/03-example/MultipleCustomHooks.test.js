import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-example/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
/**
 * para suponer que me dan algo 
 */
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');


describe('pruebas en <MultipleCustomHooks />', () => {
/**
 * debe de ponerse en cada test
 */
    // useCounter.mockReturnValue({
    //     counter: 1,
    //     increment: () => { }
    // });

    test('should de mostrarse correctamente', () => {
        useCounter.mockReturnValue({
            counter: 1,
            increment: () => { }
        });
        /**
         * esto es porque estamos usando el jest.mock
         */
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();

    });

    test('should de mostar la informacion', () => {
        useCounter.mockReturnValue({
            counter: 1,
            increment: () => { }
        });
        useFetch.mockReturnValue({
            data: [{
                author: 'emely',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb--0').text().trim()).toBe('Hola Mundo');
        expect(wrapper.find('footer').text().trim()).toBe('emely');

    });


});
