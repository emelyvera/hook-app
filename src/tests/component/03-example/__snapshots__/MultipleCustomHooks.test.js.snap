// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <MultipleCustomHooks /> should de mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</Fragment>
`;
